### PR TITLE
add <RGB LED brightness control V1.0>

### PR DESCRIPTION
**Pull request by hazemashrafali**
> ### **RGB LED brightness control V1.0**
image

### **1. Quick Links**

- [Video record](https://drive.google.com/file/d/1K0qRkgBuyhR3wP7RHdT5saFZtdDWJVqG/view?usp=sharing)
- [Design document](https://drive.google.com/file/d/1b1e0gpyKii7fHKAwIYP0z24Wv5813chz/view?usp=drive_link)
- [Test sheet](https://drive.google.com/file/d/1-Mf6FhG3ynOAUmgTUZ052KBFiNJoV3zJ/view?usp=drive_link)
- [HEX file](https://drive.google.com/file/d/1Jbs_HUwy3Q2K9vDKYyUrfQRs3wFuBGR1/view?usp=drive_link)

### **2.Hardware Components**

image

### **3. Main Application Flow**

image

### **4.Application State Flow**
image

### CODE REVIEW

File: RGB LED brightness control V1.0 Design/code/APP/app.c [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		{
+			delay_ms(TIMER1_ID,DEBOUNCE_TIME);
+			BUTTON_digitalRead(ENU_PORT_F,BUTTON_PIN,&button_value);
+			if(button_value == LOGIC_HIGH)
+			{
+				if(flag == FALSE)
+				{
+					state_flag =TRUE;
+					press_count++;
+					if(press_count == 1)
```
Comment (Hala-Abdelkader) [inline RGB LED brightness control V1.0 Design/code/APP/app.c]: You shouldn't use magic numbers

**Review by Hala-Abdelkader — CHANGES_REQUESTED** [review]
Great work

File: RGB LED brightness control V1.0 Design/code/APP/app.c (hunk 1 above)
Comment (hazemashrafali) [reply RGB LED brightness control V1.0 Design/code/APP/app.c]: DONE, you can check again after modifications.

Comment (Hala-Abdelkader) [reply RGB LED brightness control V1.0 Design/code/APP/app.c]: I re-reviewed your code and saw the requested changes and would like to thank you for your efforts.

Please note that:
-The changes you made after our review will not increase your score as we told you our comments to not do it again and lose marks in the next deliveries.

Comment (hazemashrafali) [reply RGB LED brightness control V1.0 Design/code/APP/app.c]: thank you for your support.

**Pull request merged**